File: Olympics.Service/Book/BookService.cs
using Microsoft.EntityFrameworkCore;
using Olympics.Models;
using Olympics.Models.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Olympics.Service
{
	public class BookService : BaseService, IBookService
	{
		public BookService(DataContext context): base(context) { }

		public Book GetBook(int id)
		{
			return _context.Books
				.Where(x => x.Id == id)
				.Include(x => x.Pages)
				.FirstOrDefault();
		}

		public void CreateBook(CreateBookViewModel model, User user)
		{
			Book book = new Book()
			{
				CreationUserId = user.Id,
				CreationDate = DateTime.Now,
				LastUpdatedUserId = user.Id,
				LastUpdatedDate = DateTime.Now,
				Headline = model.Headline,
				ShelveId = model.ShelveId
			};

			base.Insert<Book>(book);
			base.SaveChanges();
		}

		public void EditBook(Book book, User user)
		{
			var model = _context.Books.Where(x => x.Id == book.Id).FirstOrDefault();
			model.LastUpdatedUserId = user.Id;
			model.LastUpdatedDate = DateTime.Now;
			model.Headline = model.Headline;

			base.SaveChanges();
		}

		public void DeleteBook(int id)
		{
			var book = _context.Books.Where(x => x.Id == id).FirstOrDefault();
			_context.Books.Remove(book);
			base.SaveChanges();
		}
	}
}

File: Olympics-Frontend/src/app/shared/service/book.service.ts
import { Injectable } from '@angular/core';
import { Book } from '../models/book.model';
import { CreateBookViewModel } from '../models/createBookViewModel.model';
import { BaseService } from './base.service';

@Injectable({
  providedIn: 'root'
})
export class BookService extends BaseService {

  async getBook(id: number): Promise<Book> {
    return await super.httpGet<Book>(`book/getBook/${id}`);
  }

  createBook(model: CreateBookViewModel): void {
    super.httpPost<CreateBookViewModel>('book/createBook', model);
  }

  editBook(book: Book): void {
    super.httpPost<Book>('book/editBook', book);
  }

  deleteBook(id: number): void {
    super.httpDelete(`book/deleteBook/${id}`);
  }
}
